### DISCUSSION TITLE
You can only start an embedded section if the you consume characters at the start of the embedded block (like consuming the <style> tag)

### DISCUSSION THREAD

**Discussion by sdegutis** [microsoft/monaco-editor#3741] [Q&A]
**monaco-editor version:** 0.18.1
**Browser:** Safari 13.0.1
**OS:** macOS 10.14.6

**Steps or JS usage snippet reproducing the issue:**

First let me say, I found a workaround for this that gets the job done for me. So I don't personally need this bug fixed. I'm only mentioning it here in case it's not already known about, so that you can be made aware.

The bug is described in the Monarch documentation page:

> Bug: you can only start an embedded section if the you consume characters at the start of the embedded block (like consuming the <style> tag) (Aug 2012)

(Note that this bug says Aug 2012 so it's been 7 years.)

I can't tell if this bug still exists or not, but either the bug should be fixed if it does, or the documentation should be fixed if it doesn't.

This seems to be related to a comment in the Markdown syntax definition on the same page and on the monaco-languages repo, first found in commit 23b305079f10cb365b8badda1a57d263a31a6b41:

```js
// Note: it is tempting to rather switch to the real HTML mode instead of building our own here
// but currently there is a limitation in Monarch that prevents us from doing it: The opening
// '<' would start the HTML mode, however there is no way to jump 1 character back to let the
// HTML mode also tokenize the opening angle bracket. Thus, even though we could jump to HTML,
// we cannot correctly tokenize it in that mode yet.
```

**Comment by alexdima** [comment]
I am not 100% what the limitation is about, as that comment was most likely written by Daan Leijen, initial author of Monarch... Perhaps it would be possible to use a lookahead to match a rule and then switch to an embedded mode...

**Comment by sdegutis** [comment]
Or maybe just adding `goBack: 1` to the action would work after matching `<`? (Maybe that's what you meant by lookahead).

**Comment by NullVoxPopuli** [comment]
I'm looking at adding a language or two to monaco, as well as extending the markdown syntax.
Has anything happened with this?
